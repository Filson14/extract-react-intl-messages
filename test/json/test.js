import fs from 'fs'
import path from 'path'
import test from 'ava'
import tempy from 'tempy'
import m from '../../'

test('export json', async t => {
  const tmp = tempy.directory()
  await m(['en', 'ja'], 'test/fixtures/**/*.js', tmp)
  const en = JSON.parse(fs.readFileSync(path.resolve(tmp, 'en.json'), 'utf8'))
  const ja = JSON.parse(fs.readFileSync(path.resolve(tmp, 'ja.json'), 'utf8'))
  t.snapshot(en)
  t.snapshot(ja)
})

test('export json - nest', async t => {
  const tmp = tempy.directory()
  await m(['en', 'ja'], 'test/fixtures/**/*.js', tmp, { flat: false })
  const en = JSON.parse(fs.readFileSync(path.resolve(tmp, 'en.json'), 'utf8'))
  const ja = JSON.parse(fs.readFileSync(path.resolve(tmp, 'ja.json'), 'utf8'))
  t.snapshot(en)
  t.snapshot(ja)
})
