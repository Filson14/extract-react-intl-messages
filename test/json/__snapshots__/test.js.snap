// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export json - nest 1`] = `
Object {
  "a": Object {
    "hello": "hello",
    "world": "world",
  },
  "b": Object {
    "hello": "hello",
    "world": "world",
  },
}
`;

exports[`export json - nest 2`] = `
Object {
  "a": Object {
    "hello": "",
    "world": "",
  },
  "b": Object {
    "hello": "",
    "world": "",
  },
}
`;

exports[`export json 1`] = `
Object {
  "a.hello": "hello",
  "a.world": "world",
  "b.hello": "hello",
  "b.world": "world",
}
`;

exports[`export json 2`] = `
Object {
  "a.hello": "",
  "a.world": "",
  "b.hello": "",
  "b.world": "",
}
`;
